kythe/go/extractors/config/extractor.go

```diff
@@ -51,6 +52,59 @@ func GitCopier(repoURI string) func(ctx context.Context, outputDir string) error
 	}
 }
 
+// LocalCopier returns a function that copys a local repository.
+func LocalCopier(repoPath string) func(ctx context.Context, outputDir string) error {
+	return func(ctx context.Context, outputDir string) error {
+		gitDir := filepath.Join(repoPath, ".git")
+		// TODO(danielmoy): consider extracting all or part of this
+		// to a more common place.
+		return filepath.Walk(repoPath, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if repoPath == path {
+				// Intentionally do nothing for base dir.
+				return nil
+			}
+			if filepath.HasPrefix(path, gitDir) {
+				// Also skip all that .git stuff.
+				return nil
+			}
+			rel, err := filepath.Rel(repoPath, path)
+			if err != nil {
+				return err
+			}
+			outPath := filepath.Join(outputDir, rel)
+			if info.IsDir() {
+				if _, err := os.Stat(outPath); err != nil {
```

Comment:
`if err := os.MkdirAll(filepath.Dir(outPath), 0755); err != nil {`

Reply:
ah replacing both lines? sure that makes sense

Reply:
Yeah—`os.MkdirAll` creates all the intermediate directories, and succeeds silently if they all already exist.

The other operative point was to create only the directory, not the file. Or is `outPath` just the directory already? If so: maybe `outputBase` for the base directory and `outputDir` for the concatenated one?